### PR TITLE
fix: removed dependency on graphql

### LINKED ISSUE #1177: Graphql dependency doesn't make this a "tiny" bundle anymore
### Package

`graphql-hooks`

### Environment

- `graphql-hooks` version: latest

### Description

The latest version has a peer dependency on `graphql` package which is 40kb gzipped. It's 7 times this package's size. So in order to use 
this lightweight package we need to install 40kb additional dependencies. This destroys the whole idea of it.

### Suggested solution (optional)

Make `graphql` an optional dependency or remove the feature that requires it.

### PR DESCRIPTION
**Pull request by marceloFerreira90**
### What does this PR do?

Removed the peer dependency `graphql`
Added dependency on `@0no-co/graphql.web`
Moved typedDocumentNode to inside project (dependency on DocumentNode which is imported from` no-co/graphql `instead of `graphql`

### Related issues

closes #1177

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

File: examples/typescript/src/gql/fragment-masking.ts [hunk 1]
```diff
@@ -1,66 +1,86 @@
-import { ResultOf, DocumentTypeDecoration, TypedDocumentNode } from '@graphql-typed-document-node/core';
-import { FragmentDefinitionNode } from 'graphql';
-import { Incremental } from './graphql';
+/* eslint-disable */
+import {
+  ResultOf,
+  DocumentTypeDecoration,
+  TypedDocumentNode
+} from '@graphql-typed-document-node/core'
```
Comment (simoneb) [inline examples/typescript/src/gql/fragment-masking.ts:6]: wasn't this dependency supposed to be removed?

**Review by simoneb — COMMENTED** [review]
can you please undo all the formatting changes? they make reviewing this PR quite difficult. just disable it temporarily in your IDE

File: examples/typescript/src/gql/fragment-masking.ts (hunk 1 above)
Comment (marceloFerreira90) [reply examples/typescript/src/gql/fragment-masking.ts:6]: this is not within the graphql-hooks package itself this is the example. we still want to allow 'typedDocumentNode'  files being called here but not have a dependency inside the package itself

**Comment by marceloFerreira90** [comment]
closing this pr as my githead is not aligned with master and unable to fix it quickly new pr here: https://github.com/nearform/graphql-hooks/pull/1184

**Pull request closed without merging**